packages/cli/src/commands/upgrade/upgrade.js

```diff
@@ -60,9 +57,32 @@ const getPatch = async (currentVersion, newVersion, projectDir) => {
     return null;
   }
 
-  return patch
-    .replace(new RegExp(rnDiffAppName, 'g'), name)
-    .replace(new RegExp(rnDiffAppName.toLowerCase(), 'g'), name.toLowerCase());
+  let patchWithRenamedProjects = patch;
+
+  // rn-diff-purge only supports iOS and Android platforms
+  if (config.project.ios) {
+    patchWithRenamedProjects = patchWithRenamedProjects.replace(
+      new RegExp('RnDiffApp', 'g'),
+      // $FlowFixMe - poor typings of ProjectConfigIOST
+      config.project.ios.projectName.replace('.xcodeproj', ''),
+    );
+  }
+
+  if (config.project.android) {
```

Comment:
Maybe you would iterate on `config.platforms` and print a warning for every platform that is not android/ios? Just in case someone has `windows` integrated.